src/pages/index.tsx

```diff
@@ -19,10 +18,36 @@ import ExploreSidebar from "../components/ExploreSidebar";
 import RequestSidebar from "../components/RequestSidebar";
 import SentRequestModal from "../components/SentRequestModal";
 import ReceivedRequestModal from "../components/ReceivedRequestModal";
+import { getSession } from "next-auth/react";
 import AlreadyConnectedModal from "../components/AlreadyConnectedModal";
 
 mapboxgl.accessToken = browserEnv.NEXT_PUBLIC_MAPBOX_ACCESS_TOKEN;
 
+export async function getServerSideProps(context: GetServerSidePropsContext) {
+  const session = await getSession(context);
+
+  if (!session?.user) {
+    return {
+      redirect: {
+        destination: "/sign-in",
```

Comment:
Might be a bit extra but what are your thoughts on making these page names an enum type, just so adding more pages or changing the name of pages in the future would be a lot simpler of a process.